src/modules/production/production.cpp

```diff
@@ -49,38 +52,90 @@ namespace lean::modules {
             is_producer ? " - I'm a producer" : "");
 
     if (is_producer) {
-      auto parent_hash = block_tree_->bestBlock().hash;
-      // Produce block
-      Block block;
-      block.slot = msg->slot;
-      block.proposer_index = producer_index;
-      block.parent_root = parent_hash;
-      // block.state_root = ;
-
-      // Add a block into the block tree
-      auto res = block_tree_->addBlock(block);
-      if (res.has_error()) {
-        SL_ERROR(
-            logger_, "Could not add block to the block tree: {}", res.error());
+      fork_choice_store_->acceptNewVotes();
+
+      auto res = fork_choice_store_->produceBlock(msg->slot, producer_index);
+      if (!res.has_value()) {
+        SL_ERROR(logger_,
+                 "Failed to produce block for slot {}: {}",
+                 msg->slot,
+                 res.error());
         return;
       }
+      auto new_block = res.value();
+
+      SignedBlock new_signed_block{
+          .message = new_block, .signature = qtils::ByteArr<32>{0}
+          // signature with zero bytes for now
+      };
 
-      // Notify subscribers
-      loader_.dispatch_block_produced(std::make_shared<const Block>(block));
+      SL_INFO(logger_,
+              "Produced block for slot {} with parent {} state {}",
+              msg->slot,
+              new_block.parent_root,
+              new_signed_block.message.state_root);
 
-      // TODO(turuslan): signature
       loader_.dispatchSendSignedBlock(
-          std::make_shared<messages::SendSignedBlock>(
-              SignedBlock{.message = block}));
+          std::make_shared<messages::SendSignedBlock>(new_signed_block));
     }
   }
+  void ProductionModuleImpl::on_slot_interval_one_started(
+      std::shared_ptr<const messages::SlotIntervalOneStarted> msg) {
+    SL_INFO(logger_, "Slot interval one started on slot {}", msg->slot);
+    auto head_root = fork_choice_store_->getHead();
+    Checkpoint head{.root = head_root,
+                    .slot = fork_choice_store_->getBlockSlot(head_root)};
+    auto target = fork_choice_store_->getVoteTarget();
+    auto source = fork_choice_store_->getLatestJustified();
+    SL_INFO(logger_,
+            "For slot {}: head is {}@{}, target is {}@{}, source is {}@{}",
```

Comment:
Trivial: `slot@hash` a bit confuses, cause in means 'slot at hash'. Actually slot can have several block, not opposite. `hash@slot` would be better.